src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,361 @@
+                String r;
+                r = gui.dukeResponse("One task down sir. Here is the task I checked off:\n");
+                r = r+ gui.dukeResponse("    " + tasks.getTask(task).toString() + "\n");
+                return r;
+            } else {
+                return gui.dukeResponse("You have entered an invalid task number Sir, please input again.\n");
+            }
+        } catch (NumberFormatException e){
+            return gui.dukeResponse("Task number formatted incorrectly. Try again\n");
+        } catch (StringIndexOutOfBoundsException e){
+            return gui.dukeResponse("Enter number of the task you want to delete");
+        }
+    }
+    
+    private String delete(String userInput) {
+        try {
+            int task = Integer.parseInt(userInput.substring(7));
+            if(task > 0 && task <= tasks.getSize()){
+                Task t = tasks.deleteTask(task);
+                return echo(t.toString(), "removed");
+            } else {
+                return gui.dukeResponse("You have entered an invalid task number Sir, please input again.\n");
+            }
+        } catch (NumberFormatException e){
+            return gui.dukeResponse("Task number formatted incorrectly. Try again\n");
+        }
+    }
+    
+    private String deadline(String userInput) {
+        if (userInput.indexOf("/by")  < 11 ) {
+            return gui.dukeResponse("Enter a valid deadline activity description\n");
+        } else if (userInput.length() <= userInput.indexOf("/by") +  4) {
+            return gui.dukeResponse("Enter a valid deadline time\n");
+        } else {
+            String description = userInput.substring(9, userInput.indexOf("/by") - 1);
+            String by = userInput.substring(userInput.indexOf("/by") + 4);
+
+            try {
+                LocalDate time = LocalDate.parse(by);
+                Task t =  new Deadline(description, by);
+                tasks.addTask(t);
+                storage.addNewTask(t);
+                return echo(t.toString(), "added");
+            } catch (Exception e) {
+                return gui.dukeResponse("Enter a valid date in the format yyyy-mm-dd\n");
+            }
+        }
+    }
+    
+    private String event(String userInput) {
+        if (userInput.indexOf("/at")  < 8 ) {
+            return gui.dukeResponse("Enter a valid event activity description\n");
+        } else if (userInput.length() <= userInput.indexOf("/at") +  4) {
+            return gui.dukeResponse("Enter a valid event time\n");
+        } else {
+            String description = userInput.substring(6, userInput.indexOf("/at") - 1);
+            String at = userInput.substring(userInput.indexOf("/at") + 4);
+            Task t =  new Event(description, at);
+            tasks.addTask(t);
+            storage.addNewTask(t);
+            return echo(t.toString(), "added");
+            
+        }
+    }
+    
+    private String todo(String userInput) {
```

Comment:
Should the method name be a verb? 

Perhaps, can consider using ENUM for the Tasks todo, event, deadline? (since quite a number of code appears repeated)